### DISCUSSION TITLE
传感器采集xyz轴加速度数据和温度数据，绘图时没有显示温度数据？

### DISCUSSION THREAD

**Discussion by TTsXDU** [wh201906/SerialTest#50] [Q&A]
请问我的传感器采集到了xyz轴的加速度数据和环境的温度数据，在数据显示栏中这四项数据都显示出来了，但在绘图时只显示了xyz轴的数据，没有显示温度的。现在我想在实时绘图中显示温度曲线，请问该怎么添加呢？我试着添加数据项数到4项，但依然没有显示，是什么原因呢？我是用的是MPU6050传感器，用的是esp32作为主控板，用arduino编程。恳请指教。

**Comment by wh201906** [comment]
有数据界面和绘图界面的截图吗？

**Reply by TTsXDU** [reply]
您好这是截图，刚才discussion没有上传成功
![8A02R_QV$9{8Y5}Q0D9DO4E](https://github.com/wh201906/SerialTest/assets/121789556/2981c9be-042e-49e4-9907-c50edfa8f0d5)
![_2U3ZS$ QKP~9@P)4FJ` 99](https://github.com/wh201906/SerialTest/assets/121789556/8f1119b1-9a55-48ff-b289-1c2dba916ba6)

**Reply by TTsXDU** [reply]
您好这是截图，刚才discussion没有上传成功
![8A02R_QV$9{8Y5}Q0D9DO4E](https://github.com/wh201906/SerialTest/assets/121789556/2981c9be-042e-49e4-9907-c50edfa8f0d5)
![_2U3ZS$ QKP~9@P)4FJ` 99](https://github.com/wh201906/SerialTest/assets/121789556/8f1119b1-9a55-48ff-b289-1c2dba916ba6)

**Comment by wh201906** [comment]
程序是通过将接收到的文本以某个固定字符串分割，然后提取每一段第一个可识别的数字来绘图的。
你可以试着在` Temprature`前面加一个逗号`,`

**Reply by TTsXDU** [reply]
是在哪个文件的哪段代码添加这个逗号呢，还是在数据界面添加呢？

**Reply by wh201906** [reply]
是在发送的数据里面加
比如原先发送的是
`Accelerometer: 0.56, -0.24, 0.02m/s^2 Temperature: 29.00 °C`
改成
`Accelerometer: 0.56, -0.24, 0.02m/s^2, Temperature: 29.00 °C`

**Reply by TTsXDU** [reply]
您好，您是说在数据栏中那个”已发送“下方的”发送“那一处添加吗，但我这个都是在接收数据，没有发送数据，是这个上位机在接收传感器读取到的数据。您的意思是要在那个”发送“那里添加文本吗？
是这里吗？
![Image_1708309719472](https://github.com/wh201906/SerialTest/assets/121789556/fd33f67f-9115-4a57-9871-6e3f12497932)

**Reply by wh201906** [reply]
我的意思是修改接收到的数据，需要你修改Arduino的代码

**Reply by TTsXDU** [reply]
好的好的，明白了，感谢您的及时回复

**Reply by TTsXDU** [reply]
温度实时曲线已经能成功显示，感谢博主的帮助
